### PR TITLE
SocialLogin

### PR DESCRIPTION
**Pull request by tmh9x**

### CODE REVIEW

File: src/app/components/Icons/FacebookIcon.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+
+export default function FacebookIcon(): JSX.Element {
+  return (
+    <svg
+      id="Facebook_button"
+      data-name="Facebook button"
```
Comment (phlgr) [inline src/app/components/Icons/FacebookIcon.tsx]: This could be removed

File: src/app/components/SocialLogin/SocialLogin.tsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import classes from './SocialLogin.module.css';
```
Comment (phlgr) [inline src/app/components/SocialLogin/SocialLogin.tsx]:
```suggestion
import styles from './SocialLogin.module.css';
```

File: src/app/components/SocialLogin/SocialLogin.tsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+type SocialProps = {
```
Comment (phlgr) [inline src/app/components/SocialLogin/SocialLogin.tsx]:
```suggestion
type SocialLoginProps = {
```

File: src/app/components/SocialLogin/SocialLogin.tsx [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  title: string;
+  icon1: React.SVGProps<SVGSVGElement>;
+  icon2: React.SVGProps<SVGSVGElement>;
+};
+
+export default function SocialLogin({
+  title,
+  icon1,
+  icon2,
```
Comment (phlgr) [inline src/app/components/SocialLogin/SocialLogin.tsx]: I would expect a SocialLogin component to only have one Social Login

**Review by phlgr — CHANGES_REQUESTED** [review]

File: src/app/components/Icons/FacebookIcon.tsx [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+      <defs>
+        <linearGradient
+          id="linear-gradient"
+          x1="0.5"
+          x2="0.5"
+          y2="1"
+          gradientUnits="objectBoundingBox"
+        >
+          <stop offset="0" stop-color="#ffbb3b" />
+          <stop offset="1" stop-color="#ffa90a" />
```
Comment (lmachens) [inline src/app/components/Icons/FacebookIcon.tsx]: You can use CSS variables here. Take a look at:
https://github.com/lmachens/photoplay/pull/33/files#diff-67da3a784c1aad97057e7c3ea7fa3af53efc7e271207f791b8a2ccd6eeb43d96R23-R24

File: src/app/components/SocialLogin/SocialLogin.stories.tsx [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+export default {
+  title: 'Component/SocialLogin',
+  component: SocialLogin,
+};
+
+export const SocialLogins = (): JSX.Element => (
+  <SocialLogin
+    title="Social Logins"
+    icon1={<FacebookIcon />}
+    icon2={<GoogleIcon />}
```
Comment (lmachens) [inline src/app/components/SocialLogin/SocialLogin.stories.tsx]: I would love to see a list of social icons here to allow adding more services without changing the component.
wdyt about:
```suggestion
    icons={[{
      href: "https://facebook.com/photoplay",
      Icon: FacebookIcon
    }, {
      href: "https://google.com/photoplay",
      Icon: GoogleIcon 
    }]}
```

or:
```suggestion
    icons={[{
      href: "https://facebook.com/photoplay",
      icon: <FacebookIcon />
    }, {
      href: "https://google.com/photoplay",
      icon: <GoogleIcon />
    }]}
```

**Review by lmachens — COMMENTED** [review]
Can you update the pull request title and description to be more describable?

File: src/app/components/Icons/FacebookIcon.tsx [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+      <defs>
+        <linearGradient
+          id="linear-gradient"
+          x1="0.5"
+          x2="0.5"
+          y2="1"
+          gradientUnits="objectBoundingBox"
+        >
+          <stop offset="0" stop-color="var(--primary-color)" />
+          <stop offset="1" stop-color="#ffa90a" />
```
Comment (lmachens) [inline src/app/components/Icons/FacebookIcon.tsx]: See https://github.com/lmachens/photoplay/blob/main/src/app/globals.css#L14-L15
```suggestion
          <stop offset="0" stop-color="var(--primary-gradient-from)" />
          <stop offset="1" stop-color="var(--primary-gradient-to)" />
```

File: src/app/components/Icons/FacebookIcon.tsx [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+        </linearGradient>
+      </defs>
+      <g id="Group_5" data-name="Group 5">
```
Comment (lmachens) [inline src/app/components/Icons/FacebookIcon.tsx]: You can remove this `g` (it doesn't do anything useful)

```suggestion
```

File: src/app/components/Icons/GoogleIcon.tsx [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+          y2="1"
+          gradientUnits="objectBoundingBox"
+        >
+          <stop offset="0" stop-color="#ffbb3b" />
+          <stop offset="1" stop-color="#ffa90a" />
+        </linearGradient>
+      </defs>
+      <path
+        id="Google_Logo"
+        data-name="Google Logo"
```
Comment (lmachens) [inline src/app/components/Icons/GoogleIcon.tsx]:
```suggestion
```

File: src/app/components/SocialLogin/SocialLogin.module.css [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+  padding: 0.2em 1.5em;
+  background-color: var(--background-color);
+  margin-top: -0.8em;
+  text-align: center;
+  color: var(--text-color);
+  grid-column: 1/3;
+}
+
+.SocialLogin__link1 {
+  justify-self: end;
```
Comment (lmachens) [inline src/app/components/SocialLogin/SocialLogin.module.css]: Can you explain why this is required? I would prefer to control this in the parent element to reduce the number of classes.

File: src/app/components/SocialLogin/SocialLogin.tsx [hunk 11]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import styles from './SocialLogin.module.css';
+
+type SocialLoginProps = {
+  icon: React.SVGProps<SVGSVGElement>;
```
Comment (lmachens) [inline src/app/components/SocialLogin/SocialLogin.tsx]: You could be less specific here to allow `img` elements too:
```suggestion
  icon: JSX.Element;
```

File: src/app/components/SocialLogins/SocialLogins.tsx [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+};
+
+export default function SocialLogins({
+  socials,
+}: SocialLoginsProps): JSX.Element {
+  return (
+    <div className={styles.SocialLogin}>
+      <span className={styles.SocialLogin__title}>Social Logins</span>
+      {socials.map((social) => (
+        <SocialLogin icon={social.icon} link={social.link} />
```
Comment (lmachens) [inline src/app/components/SocialLogins/SocialLogins.tsx]: Please add a `key` to help React

**Review by lmachens — CHANGES_REQUESTED** [review]
Good job []~(￣▽￣)~*

A pull request description would be nice.

File: src/app/components/Icons/FacebookIcon.tsx (hunk 7 above)
Comment (tmh9x) [reply src/app/components/Icons/FacebookIcon.tsx]: thx

**Pull request merged**